File: app/src/main/java/com/projectcenterfvt/historicalpenza/GreetingActivity.java
package com.projectcenterfvt.historicalpenza;

import android.content.Intent;
import android.support.design.widget.TabLayout;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class GreetingActivity extends AppCompatActivity {

    ViewPager pager;
    PagerAdapter pagerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_greeting);


        pager = (ViewPager) findViewById(R.id.pager);
        pagerAdapter = new MyFragmentPagerAdapter(getSupportFragmentManager());
        pager.setAdapter(pagerAdapter);

        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabDots);
        tabLayout.setupWithViewPager(pager, true);

    }

    public void continueClick(View view) {
        startActivity(new Intent(this, MapActivity.class));
    }

}

File: app/src/main/java/com/projectcenterfvt/historicalpenza/greeting/GreetingActivity.kt
package com.projectcenterfvt.historicalpenza.greeting

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentPagerAdapter
import android.support.v7.app.AppCompatActivity
import com.projectcenterfvt.historicalpenza.R
import com.projectcenterfvt.historicalpenza.map.MapActivity
import kotlinx.android.synthetic.main.activity_greeting.*

class GreetingActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_greeting)

        pager.adapter = object : FragmentPagerAdapter(supportFragmentManager) {

            override fun getItem(position: Int): Fragment {
                return GreetingPageFragment.newInstance(position)
            }

            override fun getCount(): Int {
                return GreetingPageFragment.COUNT
            }

        }
        dots.setupWithViewPager(pager, true)

        continueButton.setOnClickListener {
            if (pager.currentItem < GreetingPageFragment.COUNT - 1) {
                pager.currentItem += 1
            } else {
                val intent = Intent(this, MapActivity::class.java)
                startActivity(intent)
                finish()
            }
        }
    }

}

